server/events/command_requirement_handler_test.go

```diff
@@ -194,6 +194,89 @@ func TestAggregateApplyRequirements_ValidateApplyProject(t *testing.T) {
 	}
 }
 
+func TestRequirements_ValidateProjectDependencies(t *testing.T) {
+	tests := []struct {
+		name        string
+		ctx         command.ProjectContext
+		setup       func(workingDir *mocks.MockWorkingDir)
+		wantFailure string
+		wantErr     assert.ErrorAssertionFunc
+	}{
+		{
+			name:    "pass no dependencies",
+			ctx:     command.ProjectContext{},
+			wantErr: assert.NoError,
+		},
+		{
+			name: "pass all dependencies applied",
+			ctx: command.ProjectContext{
+				Dependencies: []string{"project1"},
+				PullStatus: &models.PullStatus{
+					Projects: []models.ProjectStatus{
+						{
+							ProjectName: "project1",
+							Status:      models.AppliedPlanStatus,
+						},
+					},
+				},
+			},
+			wantErr: assert.NoError,
+		},
+		{
+			name: "Fail all dependencies are not applied",
+			ctx: command.ProjectContext{
+				Dependencies: []string{"project1", "project2"},
+				PullStatus: &models.PullStatus{
+					Projects: []models.ProjectStatus{
+						{
+							ProjectName: "project1",
+							Status:      models.PlannedPlanStatus,
+						},
+						{
+							ProjectName: "project2",
+							Status:      models.ErroredApplyStatus,
+						},
+					},
+				},
+			},
+			wantFailure: "Can't apply your project unless you apply its dependencies: [project1]",
+			wantErr:     assert.NoError,
+		},
+		{
+			name: "Fail one of dependencies is not applied",
+			ctx: command.ProjectContext{
+				Dependencies: []string{"project1", "project2"},
+				PullStatus: &models.PullStatus{
+					Projects: []models.ProjectStatus{
+						{
+							ProjectName: "project1",
+							Status:      models.AppliedPlanStatus,
+						},
+						{
+							ProjectName: "project2",
+							Status:      models.ErroredApplyStatus,
+						},
+					},
+				},
+			},
+			wantFailure: "Can't apply your project unless you apply its dependencies: [project2]",
+			wantErr:     assert.NoError,
```

Comment:
nitpick: Tests are good so far, can we include some more scenarios that include 3 projects to make sure Atlantis doesn't continue past the 2nd project?

Project 1: success
Project 2: fail
Project 3: skipped

Reply:
@Luay-Sol - re: the bug you were struggling with, is there a test case in here that ensures no regressions against that bug?

Reply:
oh, so the bug was related to missing 

```go
&& project.Status != models.PlannedNoChangesPlanStatus
```
to break the dependency chain